pkg/ccl/changefeedccl/alter_changefeed_stmt.go

```diff
@@ -42,6 +42,22 @@ func init() {
 
 const telemetryPath = `changefeed.alter`
 
+type telemetryAggregator struct {
+	txn *kv.Txn
+}
+
+func (a *telemetryAggregator) Count(path string) {
+	a.txn.AddCommitTrigger(func(context.Context) {
+		telemetry.Count(path)
+	})
+}
+
+func (a *telemetryAggregator) CountBucketed(path string, value int64) {
+	a.txn.AddCommitTrigger(func(context.Context) {
```

Comment:
wouldn't that add multiple triggers if txn retries?

Reply:
We clear out the transaction when we prepare for retry.  But, your comment prompted me to go look and there is a code comment that points to https://github.com/cockroachdb/cockroach/issues/18170 which indicates maybe we might have a problem still along some code paths.

Perhaps we can install a single callback and use something like sync.Once to make sure we only call it once even if it doesn't get reset.

Reply:
I've pushed a test that checks that we don't double count on a retry-like event (I rollback to a SAVEPOINT at the start of the txn).

Still not sure I love using AddCommitTrigger for this though.

Reply:
I agree re txn.AddCommitTrigger -- not a fan of it.
I was about to suggest sync.Once;
What do you think about something like https://gist.github.com/miretskiy/2800e81ea4ce779eb02a86d21e143879 instead?  Just use withTelemetryUpdate fn in alter statement

This sounds like a bigger problem though, right? Every hook based statement that updates telemetry (which is every hook based statement) will be impacted?  Do we have an issue for that?

Reply:
> This sounds like a bigger problem though, right? 

Yes, my guess is it affects all of the hook based statements.

Reply:
Made a small issue here for it: https://github.com/cockroachdb/cockroach/issues/91827

Reply:
> What do you think about something like [gist.github.com/miretskiy/2800e81ea4ce779eb02a86d21e143879](https://gist.github.com/miretskiy/2800e81ea4ce779eb02a86d21e143879) instead? Just use withTelemetryUpdate fn in alter statement

I'm not sure that works because I believe the entire plan hook gets re-run on a retry.